server/src/controllers/apps.controller.ts

```diff
@@ -107,6 +107,25 @@ export class AppsController {
     return response;
   }
 
+
+  @UseGuards(JwtAuthGuard)
+  @Post(':id/clone')
+  async clone(@Request() req, @Param() params) {
+    const existingApp = await this.appsService.find(params.id);
+    const ability = await this.appsAbilityFactory.appsActions(req.user, {});
+
+    if (!ability.can('viewApp', existingApp)) {
```

Comment:
Why the ability is `viewApp`?  🤔
Only admins and developers should be able to clone apps.

Reply:
The assumption was if the app is viewable should be able to clone them for modification as well.
Will revise then